include/cudaq/Optimizer/CodeGen/Pipelines.h

```diff
@@ -57,6 +57,9 @@ void addPipelineToQIR(mlir::PassManager &pm,
   if constexpr (QIRProfile) {
     cudaq::opt::addQIRProfilePipeline(pm, convertTo);
   }
+
+  // Make sure we don't have any leftover dead code
+  pm.addPass(mlir::createCanonicalizerPass());
```

Comment:
We need to figure out why the unrealized casts are getting inserted.

Reply:
Agreed. Would you mind looking into it? Just remove this canonicalizer and you'll see the error

Reply:
+1